src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/visits/VisitsMigrationService.kt

```diff
@@ -7,17 +7,35 @@ import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.data.MigrationCon
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.data.generateBatchId
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.Messages.MIGRATE_VISITS
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationQueueService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.NomisApiService
 
 @Service
-class VisitsMigrationService(private val queueMigrationService: MigrationQueueService) {
+class VisitsMigrationService(
+  private val queueService: MigrationQueueService,
+  private val nomisApiService: NomisApiService
+) {
   private companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun migrateVisits(migrationFilter: VisitsMigrationFilter): MigrationContext<VisitsMigrationFilter> =
-    MigrationContext(migrationId = generateBatchId(), filter = migrationFilter).apply {
-      queueMigrationService.sendMessage(MIGRATE_VISITS, this)
+  fun migrateVisits(migrationFilter: VisitsMigrationFilter): MigrationContext<VisitsMigrationFilter> {
+    val visitCount = nomisApiService.getVisits(
+      prisonIds = migrationFilter.prisonIds,
+      visitTypes = migrationFilter.visitTypes,
+      fromDateTime = migrationFilter.fromDateTime,
+      toDateTime = migrationFilter.toDateTime,
+      pageNumber = 0,
+      pageSize = 1,
```

Comment:
we only need the count at this stage so effective gets one row back but with the totals